gcloud/dns/changes.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API ResourceRecordSets."""
+
+import datetime
+
+import six
+
+from gcloud._helpers import UTC
+from gcloud._helpers import _RFC3339_MICROS
+from gcloud.exceptions import NotFound
+from gcloud.dns.resource_record_set import ResourceRecordSet
+
+
+class Changes(object):
+    """Changes are bundled additions / deletions of DNS resource records.
+
+    Changes are contained wihin a :class:`gcloud.dns.zone.ManagedZone`
+    instance.
+
+    See:
+    https://cloud.google.com/dns/api/v1/changes
+
+    :type zone: :class:`gcloud.dns.zone.ManagedZone`
+    :param zone: A zone which holds one or more record sets.
+    """
+
+    def __init__(self, zone):
+        self.zone = zone
+        self._properties = {}
+        self._additions = self._deletions = ()
+
+    @classmethod
+    def from_api_repr(cls, resource, zone):
+        """Factory:  construct a change set given its API representation
+
```

Comment:
Probably worth mentioning that this has side effects on `resource` (via the `.pop` in `_set_properties`).